### PR TITLE
feat: extend examples with comprehensive list style inheritance tests

### PR DESCRIPTION
**Pull request by Srajan-Sanjay-Saxena**
- Add list style inheritance edge cases to example.js and example-node.js
- Create example-list-style.js with advanced list type testing
- Create example-list-style-node.js for Node.js compatibility
- Add React component example-list-style.js for interactive testing
- Update React App.js to include new list style example component
- Cover conflicting styles, nested lists, custom start numbers, and bracket styles
- Test all supported list-style-type values and data-start attributes

### CODE REVIEW

File: example/react-example/src/example-list-style.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      // });
+      
+      // Create and download the file
+      // const blob = new Blob([docxBuffer], { type: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document' });
+      // const url = URL.createObjectURL(blob);
+      // const a = document.createElement('a');
+      // a.href = url;
+      // a.download = 'list-style-example.docx';
+      // a.click();
+      // URL.revokeObjectURL(url);
```
Comment (K-Kumar-01) [inline example/react-example/src/example-list-style.js]: Will this be required in future?

**Review by K-Kumar-01 — APPROVED** [review]

File: example/example.js [hunk 2]
```diff
@@ -184,6 +184,79 @@ const htmlString = `<!DOCTYPE html>
...
+            <tr>
+                <td>Bolivia</td>
+                <td>
+                    <ol>
+                        <li>Sucre</li>
+                        <li>La Paz</li>
+                    </ol>
+                </td>
+            </tr>
+        </table>
```
Comment (K-Kumar-01) [inline example/example.js]: HTML generated is not correct. Fix this

**Review by K-Kumar-01 — CHANGES_REQUESTED** [review]
Checked the code and output is not correct. Please check the logic.

File: example/example.js (hunk 2 above)
Comment (Srajan-Sanjay-Saxena) [reply example/example.js]: gotcha , there was mistake in html syntax

File: example/react-example/src/example-list-style.js (hunk 1 above)
Comment (Srajan-Sanjay-Saxena) [reply example/react-example/src/example-list-style.js]: > Will this be required in future?

might be , it shows how to integrate this backend with react

File: src/helpers/xml-builder.js [hunk 3]
```diff
@@ -566,7 +569,9 @@ const modifiedStyleAttributesBuilder = (docxDocumentInstance, vNode, attributes,
       } else if (vNodeStyleKey === 'font-family') {
         modifiedAttributes.font = docxDocumentInstance.createFont(vNodeStyleValue);
       } else if (vNodeStyleKey === 'font-size') {
-        modifiedAttributes.fontSize = fixupFontSize(vNodeStyleValue, docxDocumentInstance);
+        const convertedFontSize = fixupFontSize(vNodeStyleValue, docxDocumentInstance);
+        console.log(`[DEBUG] Processing font-size: ${vNodeStyleValue} -> ${convertedFontSize} HIP`);
```
Comment (K-Kumar-01) [inline src/helpers/xml-builder.js]: Change this log to debug.

File: src/helpers/xml-builder.js [hunk 4]
```diff
@@ -739,6 +744,9 @@ const buildFormatting = (htmlTag, options) => {
       return buildShading(options && options.color ? options.color : 'black');
     case 'fontSize':
       // does this need a unit of measure?
+      console.log(
```
Comment (K-Kumar-01) [inline src/helpers/xml-builder.js]: Same here

File: src/helpers/xml-builder.js [hunk 5]
```diff
@@ -772,6 +780,11 @@ const buildRunProperties = (attributes) => {
         options.textShadow = attributes[key];
       }
 
+      // Skip rendering strong/bold formatting if explicitly set to false
+      if (key === 'strong' && attributes[key] === false) {
+        return;
+      }
+
```
Comment (K-Kumar-01) [inline src/helpers/xml-builder.js]: Why will this be set to false explicitly? From the code I can understand what it does, better would be to explain why it does this?

File: src/helpers/xml-builder.js [hunk 6]
```diff
@@ -1115,8 +1128,11 @@ const buildRunOrHyperLink = async (vNode, attributes, docxDocumentInstance) => {
         if (!hasCustomColor) {
           modifiedAttributes.hyperlink = true;
         } else {
-          // Extract custom color from hyperlink and apply it to children
+          // Always apply hyperlink style for proper link formatting (underline, etc.)
+          modifiedAttributes.hyperlink = true;
+          // Extract custom color from hyperlink and merge with existing attributes
           const hyperlinkStyle = modifiedStyleAttributesBuilder(docxDocumentInstance, vNode, {});
```
Comment (K-Kumar-01) [inline src/helpers/xml-builder.js:1362]: Why?

File: src/helpers/xml-builder.js [hunk 7]
```diff
@@ -1115,8 +1128,11 @@ const buildRunOrHyperLink = async (vNode, attributes, docxDocumentInstance) => {
...
+          // Merge hyperlink styles while preserving existing attributes
```
Comment (K-Kumar-01) [inline src/helpers/xml-builder.js]: What is the use of this comment?

**Review by K-Kumar-01 — APPROVED** [review]
Code is working fine. PR can be merged, but please answer the questions before, and also change the console.log to console.debug

File: src/helpers/xml-builder.js (hunk 6 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js:1362]: because that's what links in html are , an underlined stuff ,  initially when i didn't add this , that underlined style was not appearing .

File: example/react-example/src/example-list-style.js (hunk 1 above)
Comment (Srajan-Sanjay-Saxena) [reply example/react-example/src/example-list-style.js]: > Will this be required in future?

although i have removed in latest commit

File: src/helpers/xml-builder.js (hunk 3 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: done in the latest commit

File: src/helpers/xml-builder.js (hunk 5 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js:787]: done in the latest commit

Comment (K-Kumar-01) [reply src/helpers/xml-builder.js]: @Srajan-Sanjay-Saxena 
In the [latest commit](https://github.com/TurboDocx/html-to-docx/pull/111/commits/4d21887761e56bd468f51eea5ea6c92b56af7fce), I see only the console.log changed to console.debug.

What was changed regarding this?

Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: > @Srajan-Sanjay-Saxena In the [latest commit](https://github.com/TurboDocx/html-to-docx/pull/111/commits/4d21887761e56bd468f51eea5ea6c92b56af7fce), I see only the console.log changed to console.debug.
> 
> What was changed regarding this?

it prevents production logs to get cluttered , and as in various above conversations , you also tolde me to change console.log to console.debug

File: src/helpers/xml-builder.js (hunk 7 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: > What is the use of this comment?

this is just a semantic description of what we are trying to do to rectify the hyperlink stye issue . Keeping the other attributes intact we are merging hyperlink styles using---
 
image

Comment (K-Kumar-01) [reply src/helpers/xml-builder.js]: @Srajan-Sanjay-Saxena 
I don't think this piece of code is required as its clear from the code what is being done here.
Please remove it.

File: src/helpers/xml-builder.js (hunk 4 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: done in the last commit

File: src/helpers/xml-builder.js (hunk 5 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: > Why will this be set to false explicitly? From the code I can understand what it does, better would be to explain why it does this?

the reason for being explicit is to prevent any unnecessary inheritance of 'strong' property from the parents , and also we are just preventing that whole attribute block from processing if the valye of that key is set to false

Comment (K-Kumar-01) [reply src/helpers/xml-builder.js]: Can you mention an example as comment in the code with explanation?
Thanks

Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: image
image

File: src/helpers/xml-builder.js (hunk 7 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: > @Srajan-Sanjay-Saxena I don't think this piece of code is required as its clear from the code what is being done here. Please remove it.
done

File: src/helpers/xml-builder.js (hunk 5 above)
Comment (Srajan-Sanjay-Saxena) [reply src/helpers/xml-builder.js]: added as a comment example inside the code in the latest commit

File: src/helpers/xml-builder.js [hunk 8]
```diff
@@ -780,7 +782,72 @@ const buildRunProperties = (attributes) => {
...
+      // • This would create <w:b/> XML element in the DOCX
+      // • Result: The span text appears BOLD in the final DOCX (WRONG!)
+      //
+      // With the check:
+      // • When attributes.strong = false, the function returns early
+      // • No buildBold() is called
+      // • No <w:b/> XML is generated
+      // • Result: The span text appears normal in the DOCX (CORRECT!)
+      // ═══════════════════════════════════════════════════════════════════════════════════════════════
       if (key === 'strong' && attributes[key] === false) {
```
Comment (K-Kumar-01) [inline src/helpers/xml-builder.js:967]: Quick question based on your explanation: What happens if the top level is italic?
